components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -57,9 +151,187 @@ bool BravePrefProvider::SetWebsiteSetting(
+                                       resource_identifier,
+                                       incognito);
+}
+
+void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
+                                          bool incognito) {
+  auto& rules = cookie_rules_[incognito];
+  rules.clear();
+
+  // add chromium cookies
+  auto chromium_cookies_iterator = PrefProvider::GetRuleIterator(
+      CONTENT_SETTINGS_TYPE_COOKIES,
+      "",
+      incognito);
+  while (chromium_cookies_iterator && chromium_cookies_iterator->HasNext()) {
+    rules.push_back(CloneRule(chromium_cookies_iterator->Next()));
+  }
+  chromium_cookies_iterator.reset();
+
+  auto brave_shields_iterator = PrefProvider::GetRuleIterator(
+      CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kBraveShields,
+      incognito);
+
+  // collect shield rules
+  std::vector<Rule> shield_rules;
+  while (brave_shields_iterator && brave_shields_iterator->HasNext()) {
+    shield_rules.push_back(CloneRule(brave_shields_iterator->Next()));
+  }
+  brave_shields_iterator.reset();
+
+  // add brave cookies after checking shield status
+  auto brave_cookies_iterator = PrefProvider::GetRuleIterator(
+      CONTENT_SETTINGS_TYPE_PLUGINS,
+      brave_shields::kCookies,
+      incognito);
+
+  auto old_rules = std::move(brave_cookie_rules_[incognito]);
+
+  while (brave_cookies_iterator && brave_cookies_iterator->HasNext()) {
+    auto rule = brave_cookies_iterator->Next();
+    bool shields_down_for_site = false;
+    if (IsActive(rule, shield_rules, &shields_down_for_site)) {
+      rules.push_back(CloneRule(rule, true));
+      brave_cookie_rules_[incognito].push_back(CloneRule(rule, true));
+    } else if (shields_down_for_site) {
+      rules.push_back(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+      brave_cookie_rules_[incognito].push_back(
+          Rule(ContentSettingsPattern::Wildcard(),
+               rule.primary_pattern,
+               ContentSettingToValue(CONTENT_SETTING_ALLOW)->Clone()));
+    }
+  }
+
+  // get the list of changes
+  std::vector<Rule> brave_cookie_updates;
+  for (const auto& new_rule : brave_cookie_rules_[incognito]) {
+    auto match = std::find_if(
+        old_rules.begin(),
+        old_rules.end(),
+        [&new_rule](const auto& old_rule) {
+          // we want an exact match here because any change to the rule
+          // is an update
+          return new_rule.primary_pattern == old_rule.primary_pattern &&
+                 new_rule.secondary_pattern == old_rule.secondary_pattern &&
+                 ValueToContentSetting(&new_rule.value) ==
+                    ValueToContentSetting(&old_rule.value);
+        });
+    if (match == old_rules.end()) {
+      brave_cookie_updates.push_back(CloneRule(new_rule));
+    }
+  }
+
+  // find any removed rules
```

Comment:
nit - everywhere in this file CC wants comments to look like sentences: `Find any removed rules.`